### PR TITLE
Add connection flakyness events and 3pp no longer detected

### PR DESCRIPTION
**Pull request by Neiva07**
https://github.com/seamapi/seam-connect/issues/4373

### CODE REVIEW

File: src/events.ts [hunk 1]
```diff
@@ -59,12 +59,21 @@ export type DeviceDeletedEvent = CommonDeviceEvent<"device.deleted">
...
+  CommonDeviceEvent<"device.third_party_integration_no_longer_detected">
+
 export type DeviceSaltoPrivacyModeActivated =
   CommonDeviceEvent<"device.salto.privacy_mode_activated">
 
 export type DeviceSaltoPrivacyModedeactivated =
   CommonDeviceEvent<"device.salto.privacy_mode_deactivated">
 
+export type DeviceConnectionBecameFlaky =
+  CommonDeviceEvent<"device.connection_became_flaky">
```
Comment (rchodava) [inline src/events.ts:72]: `device.flaky_connection_detected` or `device.unstable_connection_detected` if the pair is `device.connection_stabilized`?

**Review by rchodava — APPROVED** [review]
These look ok to me but maybe a second review just to get some consensus on naming

File: src/events.ts [hunk 2]
```diff
@@ -59,12 +59,21 @@ export type DeviceDeletedEvent = CommonDeviceEvent<"device.deleted">
 
 export type DeviceThirdPartyIntegrationDetected =
   CommonDeviceEvent<"device.third_party_integration_detected">
+export type DeviceThirdPartyIntegrationNoLongerDetected =
+  CommonDeviceEvent<"device.third_party_integration_no_longer_detected">
```
Comment (seveibar) [inline src/events.ts:63]:
```suggestion
  CommonDeviceEvent<"device.third_party_integration_removed">
```

Comment (seveibar) [reply src/events.ts:63]: CC @razor-x

File: src/events.ts (hunk 1 above)
Comment (seveibar) [reply src/events.ts:72]: i think `connection_became_flaky` is ok, `_detected` to me implies something more definitive (flaky is a judgement call on a degraded system)

Potentially `device.connection_degraded`?

Comment (seveibar) [reply src/events.ts:72]: I'm think `connection_became_flaky` is still my favorite, I would need to look up what a degraded connection is but not a flaky one.

File: src/events.ts (hunk 2 above)
Comment (seveibar) [reply src/events.ts:63]: ehh i changed my mind on this one- the pairing is nice, and we don't want to imply that we are able to detect when a third party integration is always added

**Review by seveibar — APPROVED** [review]

**Pull request merged**